### PR TITLE
Enhancement/improve-error-messaging-for-role-forbidden

### PR DESCRIPTION
**Pull request by burtenshaw**
# Description

Closes #<issue_number>

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by davidberenstein1957 — COMMENTED** [review]
I think this works great. I believe this is specifically for dataset creation but shouldn't we be covering the basic scenario of `rg.Argilla(api_url="correct", api_key="wrong")` as well?

**Comment by burtenshaw** [comment]
> I think this works great. I believe this is specifically for dataset creation but shouldn't we be covering the basic scenario of `rg.Argilla(api_url="correct", api_key="wrong")` as well?

I've improved that with changes to the message in `argilla/src/argilla/_exceptions/_api.py` . We also already have a validation message in `argilla/src/argilla/_exceptions/_client.py` that says:  ""Credentials (api_key and/or api_url) are invalid"".

**Comment by davidberenstein1957** [comment]
> > I think this works great. I believe this is specifically for dataset creation but shouldn't we be covering the basic scenario of `rg.Argilla(api_url="correct", api_key="wrong")` as well?
> 
> I've improved that with changes to the message in `argilla/src/argilla/_exceptions/_api.py` . We also already have a validation message in `argilla/src/argilla/_exceptions/_client.py` that says: ""Credentials (api_key and/or api_url) are invalid"".

but perhaps it would be nice if we could also share a similar message, urging the user to go to my-settings with their owner account?

File: argilla/src/argilla/datasets/_resource.py [hunk 1]
```diff
@@ -157,7 +162,12 @@ def create(self) -> "Dataset":
         Returns:
             Dataset: The created dataset object.
         """
-        super().create()
+        try:
+            super().create()
+        except ForbiddenError as e:
+            settings_url = f"{self._client.api_url}/user-settings"
+            message = f"User is not authorized to create a dataset in workspace {self.workspace.name}. Go to {settings_url} to view your role."
```
Comment (frascuchon) [inline argilla/src/argilla/datasets/_resource.py]: What about showing the user role from client.me?

File: argilla/src/argilla/_exceptions/_api.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ class BadRequestError(ArgillaAPIError):
 
 
 class ForbiddenError(ArgillaAPIError):
-    message = "Forbidden request to the server"
+    message = "User role is forbidden from performing this this action by server"
```
Comment (frascuchon) [inline argilla/src/argilla/_exceptions/_api.py]: What do you think about this message? The forbidden action could be raised not only when creating datasets but also when listing users (`client.users`) or getting datasets from workspaces without access (for admin users) 
```suggestion
    message = "Connected user has no privileges to perform this action on the server"
```

Comment (frascuchon) [inline argilla/src/argilla/_exceptions/_api.py]: Extra `this`
```suggestion
    message = "User role is forbidden from performing this action by server"
```

**Review by frascuchon — APPROVED** [review]

**Pull request merged**